### PR TITLE
Add Fibonacci, Exponential search

### PR DESCRIPTION
**Pull request by TheBestTvarynka**
I am creating a pull request for...

- [X] New algorithm
- [ ] Update to an algorithm
- [X] Fix an error
- [ ] Other - *Describe below*

### CODE REVIEW

File: algorithms/searches/BinarySearch.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 pub fn binary_search(array: &[i32], item: i32) -> i32 {
     let mut high: i32 = array.len() as i32 - 1;
     let mut low: i32 = 0;
-    let mut mid: i32 = 0;
+    let mut mid: i32;
```
Comment (TheBestTvarynka) [inline algorithms/searches/BinarySearch.rs:4]: This assignment gets a warning:
`warning: value assigned to 'mid' is never read`.
And I decide to fix it.

File: algorithms/searches/ExponentialSearch.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            } else {
+                break;
+            }
+        }
+        if i >= array.len() {
+            i = array.len() - 1;
+        }
+        let index = binary_search(&array[(i/2)..=i], item);
+        if index != -1 { index + (i as i32 / 2) } else { -1 }
+    }
```
Comment (TheBestTvarynka) [inline algorithms/searches/ExponentialSearch.rs:19]: Exponential search includes the Binary search itself. Code for the Binary search I take from `BinarySearch.rs`

**Comment by TheBestTvarynka** [comment]
#4

**Comment by TheBestTvarynka** [comment]
@abranhe can you review this PR? The algorithms I have written should work. I tested them (you can see tests in `main.rs` files).

**Review by abranhe — APPROVED** [review]
Thanks so much for your contribution to [The All ▲lgorithms Project](https://github.com/allalgorithms). Without people like you submitting Pull Requests we couldn't run this project. You rock!

**Pull request merged**